script.xbmc.boblight/default.py

```diff
@@ -0,0 +1,277 @@
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+'''
+import xbmc
+import xbmcaddon
+import xbmcgui
+import os
+
+__addon__      = xbmcaddon.Addon()
+__cwd__        = __addon__.getAddonInfo('path')
+__scriptname__ = __addon__.getAddonInfo('name')
+__version__    = __addon__.getAddonInfo('version')
+__icon__       = __addon__.getAddonInfo('icon')
+__ID__         = __addon__.getAddonInfo('id')
+__language__   = __addon__.getLocalizedString
+
+__profile__    = xbmc.translatePath( __addon__.getAddonInfo('profile') )
+__resource__   = xbmc.translatePath( os.path.join( __cwd__, 'resources', 'lib' ) )
+
+sys.path.append (__resource__)
+
+from settings import *
+from tools import *
+
+log( "[%s] - Version: %s Started" % (__scriptname__,__version__))
+
+capture_width  = 32
+capture_height = 32
+settings       = settings()
+capture        = xbmc.RenderCapture()
+useLegacyApi   = True
+
+class MyPlayer( xbmc.Player ):
+  def __init__(self, monitor):
+    xbmc.Player.__init__( self )
+    self.monitor = monitor
+    self.playing = False
+    log('MyPlayer - init')
+        
+  def onPlayBackStopped( self ):
+    self.playing = False
+    myPlayerChanged('stop', self.monitor)
+
+  def onPlayBackPaused( self ):
+    myPlayerChanged('paused', self.monitor)
+
+  
+  def onPlayBackEnded( self ):
+    self.playing = False
+    myPlayerChanged('stop', self.monitor)
+  
+  def onPlayBackStarted( self ):
+    self.playing = True
+    myPlayerChanged('start', self.monitor)
+    if not useLegacyApi:
+      capture.capture(capture_width, capture_height)
+  
+  def isPlaying( self ):
+    return self.playing
+
+class MyMonitor( xbmc.Monitor ):
+  def __init__( self, *args, **kwargs ):
+    xbmc.Monitor.__init__( self )
+    log('MyMonitor - init')
+        
+  def onSettingsChanged( self ):
+    settings.start()
+    if not settings.reconnect:
+      check_state(self)
+      
+  def onScreensaverDeactivated( self ):
+    settings.setScreensaver(False)
+      
+  def onScreensaverActivated( self ):    
+    settings.setScreensaver(True)
+
+class Main():
+  def __init__( self, *args, **kwargs ):
+    self.warning   = 0
+  
+  def connectBoblight(self):
+    bob.bob_set_priority(255)
+    
+    if settings.hostip == None:
+      log("connecting to local boblightd")
+    else:
+      log("connecting to boblightd %s:%s" % (settings.hostip, str(settings.hostport)))
+  
+    ret = bob.bob_connect(settings.hostip, settings.hostport)
+  
+    if not ret:
+      log("connection to boblightd failed: %s" % bob.bob_geterror())
+      if self.warning < 3 and settings.other_misc_notifications:
+        xbmc.executebuiltin("XBMC.Notification(%s,%s,%s,%s)" % (__scriptname__,
```

Comment:
this no longer works. Use xbmcgui.Dialog().notification(), see https://codedocs.xyz/xbmc/xbmc/group__python___dialog.html